src/lintsampler/lintsampler.py

```diff
@@ -288,79 +161,55 @@ def sample(self, N_samples=None, funcargs=()):
+        if N_samples:
+            u = self._generate_usamples(N_samples)
+        else:
+            u = self._generate_usamples(1)
         
         # check that the function can be evaluated at the passed-in points (i.e. needs to take the dimensions as an argument)
-
-        if self.eval_type == 'gridsample':
-
-            # evaluate the pdf
-            evalf = self._evaluate_gridded_pdf(funcargs)
+        #if self.eval_type == 'gridsample':
+        if self.ngrids == 1:
 
             # call the gridded sampler
-            X = _gridsample(*self.edgearrays,f=evalf,N_samples=N_samples,seed=self.seed,qmc=self.qmc,qmc_engine=self.qmc_engine)
-
-            return X
-
-
-        elif self.eval_type == 'freesample':
-
-            if self.dim == 1:
-
-                # pdf call needs to be 1d, so rearrange
-                corners = (np.array([self.pdf(x[0],*funcargs) for x in self.x0]),np.array([self.pdf(x[0]) for x in self.x1]))
-
-            else:
-
-                # get all combinations of starting and ending corner points (only depends on dimension)
-                combinations = self._get_startend_points()
-
-                # loop through the grids
-                evalfgrid = []
-                allcorners = []
-                for ngrid in range(0,self.ngrids):
-                    allcorners.append([]) # make a list of lists, to later concatenate
-
-                    tmpevalfgrid = self._evaluate_free_pdf(ngrid,funcargs)
-                    
-                    # assign to list
-                    evalfgrid.append(tmpevalfgrid)
-
-
-                    for combo in combinations: 
-
-                        arrslice = []
-                        for griddim,entry in enumerate(combo):
-
-                            if entry==0:
-                                arrslice.append(slice(0,self.gridshape[ngrid][griddim]-1))
-
-                            else: # must be a 1
-                                arrslice.append(slice(1,self.gridshape[ngrid][griddim]))
-
-                        allcorners[ngrid].append(evalfgrid[ngrid][tuple(arrslice)].flatten())
-
-                # now go back through and remake the corners to stack everything up
-                corners = []
-                for ncombo in range(0,len(combinations)):
-                    corners.append(np.hstack([allcorners[ngrid][ncombo] for ngrid in range(0,self.ngrids)]))
-
-
-            # do the sampling
-            X = _freesample(self.x0,self.x1,*corners,N_samples=N_samples,seed=self.seed,qmc=self.qmc,qmc_engine=self.qmc_engine)
-
-            return X
-
+            X = _grid_sample(self.grids[0], u)
 
         else:
-            raise ValueError(f"LintSampler.sample: eval_type is expected to be either gridsample or freesample.")
-
-        pass
-
-
-    def resetgrid(self,cells=()):
-        """Reset the sample grid without changing the pdf.
+            grid_masses = np.array([g.total_mass for g in self.grids])
```

Comment:
```suggestion
            # compute masses of grids and normalise
            grid_masses = np.array([g.total_mass for g in self.grids])
```
Some more comments in this section might help with maintenance later.